### PR TITLE
Add lzo

### PR DESCRIPTION
**Pull request by bshastry**
This PR:
  - Adds the lzo library to oss-fuzz (CC #402)
  - Adds target out-of-source to seek feedback from maintainer @markus-oberhumer

CC @viniul

### CODE REVIEW

File: projects/lzo/lzo_compress_target.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    /* check for an incompressible block */
+    if (out_len >= size)
+    {
+        printf("This block contains incompressible data.\n");
+        return 0;
+    }
+
+    /* Decompress. */
+    new_len = size;
+    r = lzo1x_decompress(out,out_len,in,&new_len,NULL);
```
[the remark this answers is not in the document]
Comment (bshastry) [reply projects/lzo/lzo_compress_target.c]: I agree. @markus-oberhumer What do you think of adding lzo to oss fuzz?

Would it be better to move this and future test harnesses to lzo repo? Currently this PR picks up an lzo release. Would it be better to fuzz the current development commit instead? I couldn't find a public repository for lzo unfortunately.

**Comment by bshastry** [comment]
Brainstorming here: To do this systematically, we would need to add two targets per compression method, one that compresses fuzzed data and another that decompresses it. I believe there are 10 or more compression methods in the current release version, do that would mean around 20 targets.

This is excluding other compression parameters such as the number of rounds to use etc.

Wouldn't having so many targets sit in the oss-fuzz repo lead to bit rot?

File: projects/lzo/lzo_compress_target.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply projects/lzo/lzo_compress_target.c]: Markus: we prefer to have fuzz targets upstream (less likely to break)

**Comment by jonathanmetzman** [comment]
> Brainstorming here: To do this systematically, we would need to add two targets per compression method, one that compresses fuzzed data and another that decompresses it. 

I think having these two kinds of targets are worthwhile:
```
decompress(compress(data))
```
and
```
decompress(data)
```
(decompression in the first target is controversial, since theoretically it shouldn't cover anything the second target won't, but my guess is it would be useful in practice).
>I believe there are 10 or more compression methods in the current release version, do that would mean around 20 targets.
> 
> This is excluding other compression parameters such as the number of rounds to use etc.
> 

So instead of creating one (or two) new target(s) for every method/parameter combination, you can 
take some bytes from the start of `data` to determine which compression method to use and what the parameters will be. Sort of like this:

```
index = data[0] % compression_methods.length;
method = compression_methods[index]
method(data[1], size-1)
```

> Wouldn't having so many targets sit in the oss-fuzz repo lead to bit rot?

Yes. This is probably too many.

**Comment by bshastry** [comment]
Thank you @jonathanmetzman for the discussion.

I will address these concerns on my fork and ping you back when that's ready for review.

In the meanwhile, looking forward to hearing from Markus.

**Comment by bshastry** [comment]
Ping back from Berlin :-)

@markus-oberhumer: What email address should I use as the primary contact in `project.yaml`? At the moment, it is "info@oberhumer.com" but this looks like a sales related contact end-point.

File: projects/lzo/lzo_compress_target.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (bshastry) [reply projects/lzo/lzo_compress_target.c]: @jonathanmetzman I added a `decompress` only target on the source branch of this PR. Happy for feedback.

At the moment, there is a shallow heap-buffer overrun (READ) like so for a 1 byte compressed data:

```
READ of size 8 at 0x6020000000d1 thread T0                                                             
SCARINESS: 23 (8-byte-read-heap-buffer-overflow)                                           
    #0 0x5bdcce in lzo1x_decompress /src/lzo-2.10/src/lzo1x_d.ch:120:13
```

**Comment by bshastry** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and have the pull request author add another comment and the bot will run again. If the bot doesn't comment, it means it doesn't think anything has changed.

@viniul I think you need to sign the Google CLA :-)

**Comment by viniul** [comment]
I'll make sure to sign it :)

**Comment by jonathanmetzman** [comment]
ping.
@viniul are you still planning on signing the cla?

**Comment by jonathanmetzman** [comment]
>In order to pass this check, please resolve this problem and have the pull request author add another comment and the bot will run again.

@bshastry can you comment again so that I can see if the bot updates the status?
Some of the [docs](https://opensource.google.com/docs/cla/#github-consent) imply that the status won't be updated because the pull request has multiple authors, but I guess we will find out.

**Comment by bshastry** [comment]
Bot do you think?

**Comment by viniul** [comment]
Hey @Dor1s - I signed the Google CLA for the emails I used for the commits. I think @bshastry signed his already - so we should be good to go. Can you confirm?

**Comment by Dor1s** [comment]
@viniul @bshastry thank you! Now I have to figure out how to set that `cla` label to `yes`.

**Comment by Dor1s** [comment]
Ok, as per some documentation I've found, @googlebot is not going to change the status here, and we can merge after getting the consent from all authors. Given that we have only two authors here (with a couple different git emails), I think we have a consent from everyone. Merging.

**Pull request merged**